drivers/adc/adc_ads1x1x.c

```diff
@@ -556,6 +649,70 @@ static void ads1x1x_acquisition_thread(const struct device *dev)
 	}
 }
 
+#ifdef ADC_ADS1X1X_TRIGGER
+static void ads1x1x_work_fn(struct k_work *work)
+{
+	struct ads1x1x_data *data;
+	const struct device *dev;
+
+	data = CONTAINER_OF(work, struct ads1x1x_data, work);
+	dev = data->dev;
+
+	ads1x1x_adc_perform_read(dev);
+}
+
+static void ads1x1x_conv_ready_cb(const struct device *gpio_dev,
+				      struct gpio_callback *cb,
+				      uint32_t pins)
+{
+	struct ads1x1x_data *data;
+	const struct device *dev;
+	const struct ads1x1x_config *config;
+
+	ARG_UNUSED(gpio_dev);
+
+	data = CONTAINER_OF(cb, struct ads1x1x_data, gpio_cb);
+	dev = data->dev;
+	config = dev->config;
+
+	if (config->alert_rdy.port) {
+		ads1x1x_setup_rdy_pin(dev, false);
+		ads1x1x_setup_rdy_interrupt(dev, false);
+	}
```

Comment:
Here and in other places, please consider the scenario when the GPIO/Interrupt setup/configuration fails.

Reply:
I will change that

Reply:
this is done